pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -216,17 +237,24 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		// of note, we always use the cgroupfs driver when performing this check since
 		// the input is provided in that format.
 		// this is important because we do not want any name conversion to occur.
-		cgroupManager := NewCgroupManager(subsystems, "cgroupfs")
-		if !cgroupManager.Exists(CgroupName(nodeConfig.CgroupRoot)) {
-			return nil, fmt.Errorf("invalid configuration: cgroup-root doesn't exist: %v", err)
+		if !cgroupManager.Exists(CgroupName(cgroupRoot)) {
+			return nil, fmt.Errorf("invalid configuration: cgroup-root %q doesn't exist: %v", cgroupRoot, err)
 		}
-		glog.Infof("container manager verified cgroup-root exists: %v", nodeConfig.CgroupRoot)
+		glog.Infof("container manager verified user specified cgroup-root exists: %v", cgroupRoot)
+		// Include the the top level cgroup for enforcing node allocatable into cgroup-root.
+		// This way, all sub modules can avoid having to understand the concept of node allocatable.
+		cgroupRoot = path.Join(cgroupRoot, defaultNodeAllocatableCgroupName)
```

Comment:
I thought you wanted all cgroup hierarchy changes to be tied to `cgroups-per-qos` feature? Originally, I had this the way you want it now. I think the way it is better from a rollout standpoint.
Make all cgroup changes in one shot and then apply allocatable restrictions over time.